### PR TITLE
Fixed bug when audio players didn't close in resource recordable fragment

### PR DESCRIPTION
**Pull request by mXaln**
This change is [Reviewable](https://reviewable.io/reviews/bible-translation-tools/otter/267)

### CODE REVIEW

**Review by jsarabia — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/otter/267)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @mXaln)

___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/screens/RecordableTab.kt, line 18 at r1](https://reviewable.io/reviews/bible-translation-tools/otter/267#-MYfdIY3BKdmrvppZFkG:-MYfdIY3BKdmrvppZFkH:bm48z1v) ([raw file](https://github.com/bible-translation-tools/otter/blob/c5740d2c7d553caf1ebb5ad1797c2193523e8073/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/screens/RecordableTab.kt#L18)):*
> ```Kotlin
> 
>     val recordableProperty = SimpleObjectProperty<Recordable?>()
>     private val recordResourceFragment = RecordResourceFragment(viewModel)
> ```

I'm increasingly convinced that Fragments and Views should never be created by the constructor directly, but always by the framework (specifically, the frameworks DI)

I need to better understand the problem you're trying to solve here, and the refactor may need to be somewhat substantial.

**Review by mXaln — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/otter/267)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @jsarabia)

___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/screens/RecordableTab.kt, line 18 at r1](https://reviewable.io/reviews/bible-translation-tools/otter/267#-MYfdIY3BKdmrvppZFkG:-MYj3uRa9geOItUC5rYm:b-xl6tp8) ([raw file](https://github.com/bible-translation-tools/otter/blob/c5740d2c7d553caf1ebb5ad1797c2193523e8073/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/screens/RecordableTab.kt#L18)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

I'm increasingly convinced that Fragments and Views should never be created by the constructor directly, but always by the framework (specifically, the frameworks DI)

I need to better understand the problem you're trying to solve here, and the refactor may need to be somewhat substantial.
</blockquote></details>

RecordScriptureFragment which extends RecordableFragment is docked using workspace.dock, but RecordResourceFragment which also extends RecordableFragment, is created as instance and doesn't implement onDock, onUndock methods, that's why I manually call them, in order to do stuff when this fragment is closed. That's happenning because in Resource Page we dock a fragment with tabs and then add RecordResourceFragment as a content of tab. Not sure how to refactor it then

**Review by jsarabia — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/otter/267)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @mXaln)

___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/screens/RecordableTab.kt, line 18 at r1](https://reviewable.io/reviews/bible-translation-tools/otter/267#-MYfdIY3BKdmrvppZFkG:-MYkeWKM1C97czOcE-5m:bhcuifc) ([raw file](https://github.com/bible-translation-tools/otter/blob/c5740d2c7d553caf1ebb5ad1797c2193523e8073/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/screens/RecordableTab.kt#L18)):*
<details><summary><i>Previously, mXaln (Maxim) wrote…</i></summary><blockquote>

RecordScriptureFragment which extends RecordableFragment is docked using workspace.dock, but RecordResourceFragment which also extends RecordableFragment, is created as instance and doesn't implement onDock, onUndock methods, that's why I manually call them, in order to do stuff when this fragment is closed. That's happenning because in Resource Page we dock a fragment with tabs and then add RecordResourceFragment as a content of tab. Not sure how to refactor it then 
</blockquote></details>

First up is probably to go ahead and remove the RecordableFragment and copy that code into their respective implementations (for now).

Next is that rather than calling onDock and onUndock in the bind/unbind properties methods, maybe we should just call the viewmodel.closePlayers() method?

**Review by mXaln — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/267)* status: 0 of 6 files reviewed, 1 unresolved discussion (waiting on @jsarabia)

___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/screens/RecordableTab.kt, line 18 at r1](https://reviewable.io/reviews/bible-translation-tools/orature/267#-MYfdIY3BKdmrvppZFkG:-MYpS-am1ZqRcu_GgThv:bcw8qn) ([raw file](https://github.com/bible-translation-tools/orature/blob/c5740d2c7d553caf1ebb5ad1797c2193523e8073/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/screens/RecordableTab.kt#L18)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

First up is probably to go ahead and remove the RecordableFragment and copy that code into their respective implementations (for now).

Next is that rather than calling onDock and onUndock in the bind/unbind properties methods, maybe we should just call the viewmodel.closePlayers() method?
</blockquote></details>

Done. Removed RecordableFragment, moved openPlayers and closePlayers to the viewModel, both for Scripture and Resource recordable fragments. RecordResourceFragment is still initiated by constructor, because of how tabs work.

**Review by jsarabia — CHANGES_REQUESTED** [review]
> Reviewed 1 of 6 files at r2.
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/267)* status: 1 of 6 files reviewed, 1 unresolved discussion (waiting on @mXaln)

___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/screens/RecordResourceFragment.kt, line 140 at r2](https://reviewable.io/reviews/bible-translation-tools/orature/267#-MZEi_4J-wf9KE-BQge_:-MZEi_4K2Dda9YTVobPW:b2xdo9j) ([raw file](https://github.com/bible-translation-tools/orature/blob/23b669231c8ca6905924cae5f6880faba815abbe/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/screens/RecordResourceFragment.kt#L140)):*
> ```Kotlin
>         graphic = MaterialIconView(MaterialIcon.ARROW_FORWARD, "26px")
>         action {
>             recordableViewModel.closePlayers()
> ```

this looks like it should be called from recordResourceViewModel.nextChunk()
as in, that's just part of the next/previous chunk process

same with recording a take.

basically, the ideal is that the action of a button in a view should just purely delegate to one viewmodel method.

**Review by mXaln — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/267)* status: 1 of 8 files reviewed, 1 unresolved discussion (waiting on @jsarabia)

___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/screens/RecordResourceFragment.kt, line 140 at r2](https://reviewable.io/reviews/bible-translation-tools/orature/267#-MZEi_4J-wf9KE-BQge_:-MZHjKTP-ByuFsHv3U3F:b-896fix) ([raw file](https://github.com/bible-translation-tools/orature/blob/23b669231c8ca6905924cae5f6880faba815abbe/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/screens/RecordResourceFragment.kt#L140)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

this looks like it should be called from recordResourceViewModel.nextChunk()
as in, that's just part of the next/previous chunk process

same with recording a take.

basically, the ideal is that the action of a button in a view should just purely delegate to one viewmodel method.
</blockquote></details>

Done.

**Review by jsarabia — APPROVED** [review]
:lgtm:

> Reviewed 1 of 6 files at r2.
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/267#-:-MZIt7WQ-slPomOP5eEh:bnfp4nl)* status: 2 of 8 files reviewed, all discussions resolved (waiting on @mXaln)
